main.tf

```diff
@@ -175,10 +175,10 @@ resource "google_pubsub_subscription" "push_subscriptions" {
 }
 
 resource "google_pubsub_subscription" "pull_subscriptions" {
-  for_each = var.create_topic ? { for i in var.pull_subscriptions : i.name => i } : {}
+  for_each = var.create_subscriptions ? { for i in var.pull_subscriptions : i.name => i } : {}
 
   name    = each.value.name
-  topic   = google_pubsub_topic.topic.0.name
+  topic   = var.create_topic ? google_pubsub_topic.topic.0.name : each.value.topic_name
```

Comment:
Please do not allow configuring separate topics on a per-subscription basis. At that point, there's no point to use the module at all.

All subscriptions for a single module instance must connect to the same topic.

Reply:
changed to `var.topic`